docs/use_cases/embeddings_on_browser.md

```diff
@@ -0,0 +1,628 @@
+
+In our case, we'll utilize the 'Lite' version of this model, a scaled-down and faster variant of the full model. Despite its reduced size, it maintains strong performance while demanding less computational power. This feature makes it ideal for deployment in client-side code, mobile devices, or even directly within web browsers, eliminating the need for complex installations. Furthermore, it doesn't require a dedicated GPU, making it accessible to a broader range of users.
+
+The rationale behind such models is straightforward. In many NLP applications, obtaining ample training data is a challenging endeavor. Data-hungry deep learning models are often infeasible due to this limitation, and annotating more supervised training data is an expensive solution. Consequently, most NLP projects in research and industry contexts can only access relatively small training datasets.
+
+To address this constraint, many models employ pre-trained word embeddings like word2vec or GloVe, which transform individual words into vectors. However, recent developments have shown that pre-trained sentence-level embeddings can deliver impressive performance.
+
+The Universal Sentence Encoder excels at generating embeddings for complete English sentences. When you input an English string, it generates a fixed-length vector embedding representing the entire sentence. These sentence embeddings are highly effective for computing semantic similarity between sentences and have demonstrated excellent performance in various semantic textual similarity benchmarks.
+
+Moreover, these sentence embeddings are flexible and can be fine-tuned for specific tasks. Even with minimal task-specific training data, they can achieve surprisingly good results.
+
+The model is built upon the transformer architecture, leveraging the attention mechanism to create context-aware representations for each word in a sentence. It carefully considers the order and identity of all other terms. These word representations are then combined into a fixed-length sentence vector. This combination is achieved by element-wise summation, followed by division by the square root of the sentence length. This normalization process prevents shorter sentences from dominating solely due to their brevity.
+
+The primary objective is to create the encoder as a versatile tool suitable for various tasks. The model achieves this through multi-task learning, training a single model to support multiple downstream tasks.
+
+In summary, the Universal Sentence Encoder excels at generating embeddings that represent the holistic meaning of sentences, even when dealing with limited training data, making it a valuable resource for NLP tasks.
+
+[](https://arxiv.org/pdf/1803.11175.pdf)
+
+# Our step-by-step tutorial
+
+## Import modules
```

Comment:
would it be possible to remove some of these comments entirely and the few critical ones that remain, move them into the code snippet itself to keep everything in 1 place a bit more? (otherwise the reader has to jump between this and the code snippet below)